### ISSUE TITLE
Sereal 4.006 does not export read_sereal and decode_sereal in contrast to the documentation

### ISSUE THREAD

**Issue by ppisar** [Sereal/Sereal#197]
Sereal 4.006 stopped exporting read_sereal and decode_sereal symbols although it still documents them in the POD:

~~~~
=head1 NAME

Sereal - Fast, compact, powerful binary (de-)serialization

=head1 SYNOPSIS

    use Sereal qw(
      get_sereal_decoder
      get_sereal_encoder
      clear_sereal_object_cache
      encode_sereal
      decode_sereal
      write_sereal
      read_sereal
~~~~

~~~~
$ perl -Ilib -e 'use Sereal qw(read_sereal);'
"read_sereal" is not exported by the Sereal module
Can't continue after import errors at -e line 1.
BEGIN failed--compilation aborted at -e line 1.
~~~~

This used to work in 4.005:

~~~~
[test@fedora-31 Sereal-4.005]$ perl -Ilib -e 'use Sereal qw(read_sereal);'
~~~~

Is the removal of the export intentional? If it is, the documentation should be updated. By the way, this is an API breaking change and it should be properly documented in the Changes file that's missing any entry for 4.006.

**Comment by demerphq** [comment]
On Tue, 9 Apr 2019 at 15:23, Petr Pisar <notifications@github.com> wrote:

> Sereal 4.006 stopped exporting read_sereal and decode_sereal symbols
> although it still documents them in the POD:
>
> =head1 NAME
>
> Sereal - Fast, compact, powerful binary (de-)serialization
>
> =head1 SYNOPSIS
>
>     use Sereal qw(
>       get_sereal_decoder
>       get_sereal_encoder
>       clear_sereal_object_cache
>       encode_sereal
>       decode_sereal
>       write_sereal
>       read_sereal
>
> $ perl -Ilib -e 'use Sereal qw(read_sereal);'
> "read_sereal" is not exported by the Sereal module
> Can't continue after import errors at -e line 1.
> BEGIN failed--compilation aborted at -e line 1.
>
> This used to work in 4.005:
>
> [test@fedora-31 Sereal-4.005]$ perl -Ilib -e 'use Sereal qw(read_sereal);'
>
> Is the removal of the export intentional? If it is, the documentation
> should be updated. By the way, this is an API breaking change and it should
> be properly documented in the Changes file that's missing any entry for
> 4.006.
>
No, it was a daft mistake. Thank you so much for catching it and reporting
it so quickly.

I will roll a release shortly, right after I add tests to make sure all
these exports work as intended.

Thanks a lot. Really.

Yves

-- 
perl -Mre=debug -e "/just|another|perl|hacker/"

**Comment by demerphq** [comment]
This should be fixed in v4.007

Thanks!

Yves

On Tue, 9 Apr 2019 at 15:23, Petr Pisar <notifications@github.com> wrote:
>
> Sereal 4.006 stopped exporting read_sereal and decode_sereal symbols although it still documents them in the POD:
>
> =head1 NAME
>
> Sereal - Fast, compact, powerful binary (de-)serialization
>
> =head1 SYNOPSIS
>
>     use Sereal qw(
>       get_sereal_decoder
>       get_sereal_encoder
>       clear_sereal_object_cache
>       encode_sereal
>       decode_sereal
>       write_sereal
>       read_sereal
>
> $ perl -Ilib -e 'use Sereal qw(read_sereal);'
> "read_sereal" is not exported by the Sereal module
> Can't continue after import errors at -e line 1.
> BEGIN failed--compilation aborted at -e line 1.
>
> This used to work in 4.005:
>
> [test@fedora-31 Sereal-4.005]$ perl -Ilib -e 'use Sereal qw(read_sereal);'
>
> Is the removal of the export intentional? If it is, the documentation should be updated. By the way, this is an API breaking change and it should be properly documented in the Changes file that's missing any entry for 4.006.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or mute the thread.



-- 
perl -Mre=debug -e "/just|another|perl|hacker/"

**Comment by ppisar** [comment]
I confirm 4.007 fixed this issue.
